src/sync.cpp

```diff
@@ -143,14 +143,17 @@ static void push_lock(void* c, const CLockLocation& locklocation)
             break;
 
         const LockPair p1 = std::make_pair(i.first, c);
+        const LockPair p2 = std::make_pair(c, i.first);
+        if (lockdata.lockorders.count(p2)) {
```

Comment:
Any reason this is now being done before `if (lockdata.lockorders.count(p1))continue`?

Reply:
[Updated](https://github.com/bitcoin/bitcoin/pull/19340#issuecomment-667676282).